app/src/main/java/com/android/bookswap/model/chat/FirebaseMessagingService.kt

```diff
@@ -0,0 +1,113 @@
+package com.android.bookswap.model.chat
+
+import android.app.NotificationChannel
+import android.app.NotificationManager
+import android.app.PendingIntent
+import android.content.Intent
+import android.os.Build
+import androidx.core.app.NotificationCompat
+import com.android.bookswap.MainActivity
+import com.android.bookswap.R
+import com.google.firebase.messaging.FirebaseMessagingService
+import com.google.firebase.messaging.RemoteMessage
+
+/** Service to handle Firebase Cloud Messaging notifications. */
+open class MyFirebaseMessagingService : FirebaseMessagingService() {
+  /**
+   * Called when a message is received.
+   *
+   * @param remoteMessage the message received from Firebase Cloud Messaging.
+   */
+  override fun onMessageReceived(remoteMessage: RemoteMessage) {
+    val notification = remoteMessage.notification
+    val title = notification?.title ?: "Hello There!"
+    val body = notification?.body ?: "You have a new notification"
+    sendNotification(title, body)
+  }
+
+  /**
+   * Sends a notification to the user.
+   *
+   * @param title the title of the notification.
+   * @param messageBody the body of the notification.
+   */
+  fun sendNotification(title: String, messageBody: String) {
+    // Create an intent to open the MainActivity when the notification is clicked
+    val intent =
+        Intent(this, MainActivity::class.java).apply { addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP) }
+    val pendingIntent =
```

Comment:
The request code is the same for both 'accept' and 'decline' (request code = 0). It would be better to use two different request codes to avoid potential conflicts. You can refer to this [StackOverflow thread](https://stackoverflow.com/questions/21526319/whats-requestcode-used-for-on-pendingintent) for more information on the issues that may arise.

Reply:
And you can also create Constant variable for the number choose Ex: REQUEST_CODE_ACCEPT